percona/controller/pgcluster/controller.go

```diff
@@ -535,6 +539,47 @@ func (r *PGClusterReconciler) reconcileCustomExtensions(cr *v2.PerconaPGCluster)
 		extensionKeys = append(extensionKeys, key)
 	}
 
+	if cr.CompareVersion("2.6.0") >= 0 {
+		var extensionKeysForDeletion []string
+		var extensionsKeysAnnotations []string
+
+		if cr.Spec.Metadata.Annotations != nil {
+			if val, ok := cr.Spec.Metadata.Annotations["custom_extensions_list"]; ok && val != "" {
+				extensionsKeysAnnotations = strings.Split(val, ",")
+			}
+		}
+
+		for i := range extensionsKeysAnnotations {
+			extensionsKeysAnnotations[i] = strings.TrimSpace(extensionsKeysAnnotations[i])
+		}
+		annotationExtensionsMap := make(map[string]struct{})
+
+		crExtensionsMap := make(map[string]struct{})
+		// extensions added to annotations
+		for _, ext := range extensionsKeysAnnotations {
+			annotationExtensionsMap[ext] = struct{}{}
+		}
+
+		// Check for missing entries in crExtensions
+		for _, ext := range extensionsKeysAnnotations {
+			// If an object exists in extensionsKeysAnnotations but not in crExtensions, the extension should be deleted.
+			if _, exists := crExtensionsMap[ext]; !exists {
+				extensionKeysForDeletion = append(extensionKeysForDeletion, ext)
+			}
+		}
+
+		if len(extensionKeysForDeletion) > 0 {
+			var exec postgres.Executor
+			err := extensions.DisableCustomExtensionsInPostgreSQL(ctx, extensionKeysForDeletion, exec)
+			if err != nil {
+				return errors.Wrap(err, "custom extension deletion")
+			}
+		}
+
+		cr.Spec.Metadata.Annotations["custom_extensions_list"] = strings.Join(extensionKeys, ", ")
```

Comment:
You can join only with a comma (`,`), no need for `, ` (comma and space), that way you don't need to do `strings.Trim`.